### ISSUE TITLE
The FAHViewer button in FAHControl can only view the Demo protein

### ISSUE THREAD

**Issue by bb30994** [FoldingAtHome/fah-control#31]
There needs to be an option associated with the FAHViewer button that allows the user to select an active WU/slot to provide real-time images.

**Comment by bb30994** [comment]
I believe this problem is still present on Windows and Mac OS_X v7.5.1  It seems to work correctly only in Linux.

**Comment by gynnantonix** [comment]
I'm running OSX 10.14.3 and client 7.5.1, and the viewer is working with both local and remote clients.  It's not working under Windows 10 (client 7.5.1), though.

**Comment by jcoffland** [comment]
The Windows problem is most likely a graphics driver issue.  You need OpenGL support.  It won't work in a VM.  Try upgrading your graphics driver.

**Comment by gynnantonix** [comment]
If it were a driver issue then the viewer wouldn't be able to render the
demo molecule.  This is a network issue.

**Comment by bb30994** [comment]
No, jcoffland, it is NOT a driver problem.  Running the stand-alone viewer will work PROVIDED you start it in the right directory.  In various FAHCores, the viewer data seems to have been moved from one place to another ... plus I have to be able to point to the data from each of the active WUs from inside of FAHControl.

FAHViewer and FAHCore have to use the same file-naming convention.  (... or if you prefer, you can call it a network issue, since I suppose it should also work on any of the FAHClients accessible in FAHControl even if they don't happen to be on 127.0.0.1:36330.)

**Comment by foo25amc** [comment]
I can confirm the same issue on Windows 10. Works absolutely fine on my rented server in Paris running Ubuntu 18.04 via Openbox and X2Go.

**Comment by Guenni75** [comment]
The problem is still there on Win10.
I tried it today and the Viewer only runs a Demo with no connection.
Graphic driver updates yesterday!!

**Comment by waded** [comment]
FAHViewer only showing the no-connection/demo protein was my experience also with 7.5.1 on Windows 10 18363.720. I tried the workaround I've seen elsewhere of replacing FAHViewer executable and some other DLL (can't recall which) from 7.5.1 with that from 7.4.4, and can confirm that makes a difference on this PC. 7.4.4 appears to work correctly.

**Comment by ppericard** [comment]
Same probleme here. On Win10 with recent AMD CPU and GPU, with up to date drivers.

**Comment by waded** [comment]
I recommend closing this issue, as the behavior's not specific fah-control (it also happens when launching FAHViewer from e.g. Windows Start menu directly), and the details are better represented by  FoldingAtHome/fah-issues#1283. FoldingAtHome/fah-issues#1283 is also pointed to in a number of forum posts as being a very well known matter.

**Comment by waded** [comment]
I've deleted a couple of my comments that were off base.

What's being asked for here are separate buttons in FAHClient for each slot, so that FAHViewer can be launched directly to see the visualization for that slot.

That there's a bug in FAHViewer 7.5.1 where it displays Demo protein due to a connectivity issue between FAHClient and FAHViewer (https://github.com/FoldingAtHome/fah-issues/issues/1283) is a different matter. That one can be worked around somewhat by running FAHViewer distributed in version 7.4.4, as is sometimes discussed on the forums.

**Comment by alfanhui** [comment]
I have detailed instrustions on how to fix the viewer using the older 7.4.4 version here: 
https://gist.github.com/alfanhui/fd69de8354047609fad5ebbef829eb94

**Comment by shorttack** [comment]
Duplicate [https://github.com/FoldingAtHome/fah-issues/issues/1283](url)

**Comment by cachely** [comment]
> I've deleted a couple of my comments that were off base.
> 
> What's being asked for here is a way to launch FAHViewer from FAHControl to see visualization for a specific slot. (e.g. a button in FAHControl per slot, rather than one button for all slots.)
> 
> That there's a bug in FAHViewer 7.5.1 where it errantly displays Demo protein due to a connectivity issue ([FoldingAtHome/fah-issues#1283](https://github.com/FoldingAtHome/fah-issues/issues/1283)) is a different matter. That can be worked around somewhat by running FAHViewer distributed in version 7.4.4, as is discussed there and in forums. Once running FAHViewer 7.4.4, you should be able to, as something of a workaround, using the [ and ] keys to switch slots.

This worked for me. Thank you!

**Comment by shorttack** [comment]
1. I have reports that the Viewer works with the beta, found at [https://foldingathome.org/beta/](url). If you want Viewer, try the beta. It will show you current protein.
2. Recategorized from Duplicate to Enhancement. See original request: "There needs to be an option associated with the FAHViewer button that allows the user to select an active WU/slot to provide real-time images." My functional interpretation of the proposed enhancement is "add a button (or right-click) to invoke FAHviewer for the selected slot."

**Comment by waded** [comment]
> My functional interpretation of the proposed enhancement is "add a button (or right-click) to invoke FAHviewer for the selected slot."

Agree. Consider rewriting the title of this issue to that effect to focus the work/commentary.

**Comment by bb30994** [comment]
> Duplicate https://github.com/FoldingAtHome/fah-issues/issues/1283
That issue is 404.  Where is the one you intended?

**Comment by KEINOS** [comment]
Same issue. On Windows10, FAHViewer shows only demo protein **even launching it directly** but FAHControl seems working well. This doesn't happen on macOS Mojave.

- FAH Client: 7.5.1
  - Install EXE: `fah-installer_7.5.1_x86.exe`
- Windows10 Pro 64bit (Lang: Japanese)
    - Build: 19041.207
    - Windows Feature Experience Pack: 120.2202.130.0
    - RAM: 16GB
    - CPU: Intel Pentium Silver N5000 @ 1.10GHz
    - GPU: Intel UHD Graphics 605

Unfortunately, @alfanhui 's workaround didn't fix my problem. 😭

Though, the **beta version v7.6.9 did fix my issue**. Thx @shorttack 
- https://foldingathome.org/beta/
    - Install EXE: `fah-installer_7.6.9_x86.exe`
    - Note: The Windows Defender SmartScreen will block the installation. Click the "detail info" link to proceed. It will warn you several times, therefore it is not recommended to let worriers/amateurs install the beta.
- Index of the available installer for Win10: 
        - https://download.foldingathome.org/releases/public/release/fah-installer/windows-10-32bit/

**Comment by PantherX** [comment]
@KEINOS 
Please note that 7.6.9 is now considered a stable release :)

**Comment by KEINOS** [comment]
@PantherX 
Indeed! Thank you!
